### PR TITLE
Features: support Cocopod

### PR DESCRIPTION
**Pull request by Ali72**
features:
*support Cocopod 
*prevent crashes on the simulators

To install with cocoa pods, you have to install ExceptionCatcher from [this](https://github.com/Ali72/ExceptionCatcher) fork inside the pod file in the root project.

I also send [this](https://github.com/sindresorhus/ExceptionCatcher/pull/5) pull request to the ExceptionCatcher owner after he accepts that you can remove my fork from your root project.

Please accept this feature that we need in our project.

### CODE REVIEW

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>BuildableError(message: "Some model objects have unexpectedly changed since the build was started. Please try again.", buildable: <IDESwiftPackageBundleProductBuildable:ObjectIdentifier(0x000060000191b510):'<invalidated>'>)</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Testing cancelled because the build failed.</td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/7c603cb6-a7df-4aa4-b3fd-0eee33307d5b" target="_blank">Bitrise</a>
  </td>
    </tr>
  </tbody>
</table>

## Code Coverage Report
| Name | Coverage ||
| --- | --- | --- |


### SwiftLint found issues

| Severity | File | Reason |
| -------- | ---- | ------ |
Warning | Diagnostics.swift:18 | Types used for hosting only static members should be implemented as a caseless enum to avoid instantiation (`convenience_type`) |
Warning | DiagnosticsReporter.swift:8 | The disabled 'line_length' rule should be re-enabled before the end of the file (`blanket_disable_command`) |
Warning | DiagnosticsReporter.swift:169 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (`colon`) |
Warning | DiagnosticsReporter.swift:169 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (`colon`) |
Warning | DiagnosticsReporter.swift:169 | There should be no space before and one after any comma (`comma`) |
Warning | DiagnosticsReporter.swift:169 | Return arrow and return type should be separated by a single space or on a separate line (`return_arrow_whitespace`) |
Warning | DiagnosticsReporter.swift:168 | Lines should not have trailing whitespace (`trailing_whitespace`) |
Warning | DiagnosticsReporter.swift:170 | Lines should not have trailing whitespace (`trailing_whitespace`) |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against aeb778bacd7797aa3a1e6c49202400f626bea0a7
</p>

File: Sources/Logging/DiagnosticsLogger.swift [hunk 1]
```diff
@@ -257,9 +257,7 @@ private extension DiagnosticsLogger {
                 autoreleasepool {
                     outputPipe.fileHandleForWriting.write(data)
 
-                    guard let string = String(data: data, encoding: .utf8) else {
-                        return assertionFailure("Invalid data is logged")
```
Comment (AvdLee) [inline Sources/Logging/DiagnosticsLogger.swift:261]: I'd rather keep this assertion failure in place for debugging purposes. Note that assertions only fire during debugging, not for production builds.

**Comment by AvdLee** [comment]
@Ali72 thanks for your contribution. Unfortunately, we decided to move away from CocoaPods and we don't want to be the maintainers of it anymore. We would have to ensure it keeps working and we keep pushing new versions for it. 

Therefore, I'm going to close this PR. Like Bas mentioned, feel free to use your own fork for internal CocoaPods distribution.

I've looked at the rest of the code, but I could not find any code changes that fixes crashes. If you do find improvements that solves potentially crashes, feel free to open a separate PR for these. 

Thanks!

File: Sources/Logging/DiagnosticsLogger.swift (hunk 1 above)
Comment (Ali72) [reply Sources/Logging/DiagnosticsLogger.swift:261]: yes , and also crash when using simulators

Comment (AvdLee) [reply Sources/Logging/DiagnosticsLogger.swift:261]: It does in the Simulator, which is great as it can help you fix logging issues

**Pull request closed without merging**